lambdas/check-existing-identity/src/main/java/uk/gov/di/ipv/core/checkexistingidentity/CheckExistingIdentityHandler.java

```diff
@@ -476,6 +482,34 @@ private JourneyResponse buildF2FNoMatchResponse(
         return JOURNEY_F2F_FAIL;
     }
 
+    private JourneyResponse buildDCMAWContinuationResponse(
+            VerifiableCredentialBundle credentialBundle,
+            Vot lowestGpg45ConfidenceRequested,
+            ClientOAuthSessionItem clientOAuthSessionItem,
+            AuditEventUser auditEventUser)
+            throws ClientOauthSessionNotFoundException, IpvSessionNotFoundException,
+                    VerifiableCredentialException {
+        var criResponseItem =
+                criResponseService.getCriResponseItem(
+                        clientOAuthSessionItem.getUserId(), DCMAW_ASYNC);
+        var previousIpvSessionItem =
+                ipvSessionService.getIpvSessionByCriOAuthSessionId(criResponseItem.getOauthState());
```

Comment:
This `criOAuthSessionId` is already a Global Secondary Index of the IPV session table. This is O(log(N)).